packages/scene-composer/src/hooks/useMatterportTags.ts

```diff
@@ -164,31 +165,61 @@ const updateTag = async (
       return;
     }
 
-    const tagComp = tagIndex !== -1 ? updateTagEntityComponent(components[tagIndex]) : undefined;
+    const isUpdate = !!node.properties.layerIds?.at(0);
+    const tagComp =
+      tagIndex !== -1
+        ? (isUpdate ? updateTagEntityComponent : createTagEntityComponent)(components[tagIndex])
+        : undefined;
     const overlayComp =
       dataOverlayIndex !== -1
-        ? updateOverlayEntityComponent(components[dataOverlayIndex] as IDataOverlayComponent)
+        ? (isUpdate ? updateOverlayEntityComponent : createOverlayEntityComponent)(
+            components[dataOverlayIndex] as IDataOverlayComponent,
+          )
         : undefined;
-    const nodeComp = updateNodeEntityComponent(node, layerId);
+    const nodeComp = (isUpdate ? updateNodeEntityComponent : createNodeEntityComponent)(node, layerId);
 
-    const updateEntity: UpdateEntityCommandInput = {
-      workspaceId: undefined,
-      entityId: ref,
-      entityName: item.label + '_' + ref,
-      componentUpdates: {
-        Node: nodeComp,
-      },
-    };
-    if (tagComp) {
-      updateEntity.componentUpdates![KnownComponentType.Tag] = tagComp;
-    }
-    if (overlayComp) {
-      updateEntity.componentUpdates![KnownComponentType.DataOverlay] = overlayComp;
-    }
-    try {
-      await sceneMetadataModule.updateSceneEntity(updateEntity);
-    } catch (e) {
-      console.error('Update scene node entity failed', e);
+    if (isUpdate) {
+      const updateEntity: UpdateEntityCommandInput = {
+        workspaceId: undefined,
+        entityId: ref,
+        entityName: item.label + '_' + ref,
+        componentUpdates: {
+          Node: nodeComp,
+        },
+      };
+      if (tagComp) {
+        updateEntity.componentUpdates![KnownComponentType.Tag] = tagComp;
+      }
+      if (overlayComp) {
+        updateEntity.componentUpdates![KnownComponentType.DataOverlay] = overlayComp;
+      }
+
+      try {
+        await sceneMetadataModule.updateSceneEntity(updateEntity);
+      } catch (e) {
+        console.error('Update scene node entity failed', e);
+      }
+    } else {
+      const createEntity: CreateEntityCommandInput = {
+        workspaceId: undefined,
+        entityId: ref,
+        entityName: item.label + '_' + ref,
+        parentEntityId: sceneRootId,
+        components: {
+          Node: nodeComp,
+        },
+      };
+      if (tagComp) {
```

Comment:
Maybe you can make them as switch statement for future use?

Reply:
there isn't a thing to switch on